.github/workflows/build-metrics.yml

```diff
@@ -116,3 +116,54 @@ jobs:
             --pr_number "${{ github.event.number }}" \
             --sha "${{ inputs.ref || github.sha }}" \
             "/tmp/metrics"
+
+  upload-report:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    name: Generate and Upload Build Metric Report
+    needs: metrics
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

Comment:
Any reason why we need checkout at this particular sha ? The job above just uses checkout@4 ?

Reply:
This is pinned to a specific version of the action due to the required `contents:write` permission.